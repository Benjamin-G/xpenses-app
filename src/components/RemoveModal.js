import React from 'react'
import Modal from 'react-modal'

const RemoveModal = (props) => (
  <Modal isOpen={props.isOpen}
         onRequestClose={props.onDenyClose}
         contentLabel="Selected Option"
         closeTimeoutMS={333}
         ariaHideApp={false}
         className="modal">
    <h3 className="modal__title">Would you like to remove this expense?</h3>
    <div>
      <button className="button button--modal" onClick={props.onConfirmClose}>Confirm</button>
      <button className="button" onClick={props.onDenyClose}>Deny</button>
    </div>
  </Modal>
)

export default RemoveModal