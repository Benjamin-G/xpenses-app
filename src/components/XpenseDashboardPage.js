import React from 'react'

import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const XpenseDashboardPage = () => (
  <div>
    <ExpensesSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
)

export default XpenseDashboardPage
