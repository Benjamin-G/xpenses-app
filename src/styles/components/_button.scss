.button {
  background-color: $blue;
  border:none;
  color:#FFF;
  cursor:pointer;
  display: inline-block;
  font-size: $font-size-large;
  font-weight: 300;
  line-height: 1;
  padding:$s-size;
  text-decoration: none;
}

.button--link {
  background: none;
}

.button--secondary {
  background-color: #888;
  margin-bottom: $m-size;
}

.button--login {
  margin-top: $s-size;
}
