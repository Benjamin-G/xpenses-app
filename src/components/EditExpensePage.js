import React from 'react'
import { connect } from 'react-redux'

import ExpenseForm from './ExpenseForm'
import RemoveModal from './RemoveModal'
import { startEditExpense , startRemoveExpense } from '../actions/expenses'

export class EditExpensePage extends React.Component {
  state = {
    isModalOpen: false
  }
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }
  openModal = () =>{
    this.setState({isModalOpen: true})
  }
  handleConfirmedRemove = () => {
    this.setState({isModalOpen: false})
    this.props.removeExpense({ id : this.props.expense.id })
    this.props.history.push('/')
  }
  handleDeniedRemove = () => {
    this.setState({isModalOpen: false})
  }

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit}/>
          
          <button className="button button--secondary" onClick={this.openModal}>Remove Expense</button>
          <RemoveModal isOpen={this.state.isModalOpen} onConfirmClose={this.handleConfirmedRemove}
          onDenyClose={this.handleDeniedRemove} />
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  editExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  removeExpense: (data) => dispatch(startRemoveExpense(data))
})

const mapStateToProps = (state , props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)
