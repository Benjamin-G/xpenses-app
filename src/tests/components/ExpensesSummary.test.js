import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should correctly render ExpensesSumary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should correctly render ExpensesSumary with many expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={5} expensesTotal={12345}/>)
  expect(wrapper).toMatchSnapshot()
})
