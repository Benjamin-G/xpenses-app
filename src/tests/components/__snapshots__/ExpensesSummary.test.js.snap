// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSumary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h1>
</div>
`;

exports[`should correctly render ExpensesSumary with many expense 1`] = `
<div>
  <h1>
    Viewing 
    5
     
    expenses
     totalling 
    $123.45
  </h1>
</div>
`;
